code/media/src/test/java/com/adobe/marketing/mobile/media/internal/MediaSessionTests.java

```diff
@@ -521,4 +521,46 @@ public void test_process_NonSessionStartEvent_sessionCreateShouldNotDispatched()
         assertTrue(didSendNetworkRequest());
         assertFalse(mockDispatcherSessionCreated.dispatchSessionCreatedWasCalled);
     }
+
+    @Test
+    public void test_process_retry_connectionIsNull() throws Exception {
```

Comment:
could we test something similar for offline media sessions? looks like we might need to consolidate a few of the existing tests, e.g. test_reportCompletedSessions_nullConnection